aiida_quantumespresso/workflows/protocols/utils.py

```diff
@@ -0,0 +1,125 @@
+        """
+        data = load_protocol_file(cls)
+        return {protocol: {'description': values['description']} for protocol, values in data['protocols'].items()}
+
+    @classmethod
+    def get_protocol_inputs(cls, protocol=None, overrides=None):
+        """Return the inputs for the given workflow class and protocol.
+
+        :param cls: the workflow class.
+        :param protocol: optional specific protocol, if not specified, the default will be used
+        :param overrides: dictionary of inputs that should override those specified by the protocol. The mapping should
+            maintain the exact same nesting structure as the input port namespace of the corresponding workflow class.
+        :return: mapping of inputs to be used for the workflow class.
+        """
+        data = load_protocol_file(cls)
+        protocol = protocol or data['default_protocol']
+        inputs = recursive_merge(data['default_inputs'], data['protocols'][protocol])
+        inputs.pop('description')
+
+        if overrides:
+            return recursive_merge(inputs, overrides)
+
+        return inputs
+
+
+def recursive_merge(left, right):
+    """Recursively merge two dictionaries into a single dictionary.
+
+    If any key is present in both ``left`` and ``right`` dictionaries, the value from the ``right`` dictionary is
+    assigned to the key.
+
+    :param left: first dictionary
+    :param right: second dictionary
+    :return: the recursively merged dictionary
+    """
+    import collections
+
+    for key, value in left.items():
+        if key in right:
+            if isinstance(value, collections.abc.Mapping) and isinstance(right[key], collections.abc.Mapping):
+                right[key] = recursive_merge(value, right[key])
+
+    merged = left.copy()
+    merged.update(right)
+
+    return merged
+
+
+def load_protocol_file(cls):
+    """Load the protocol file for the given workflow class.
+
+    :param cls: the workflow class.
+    :return: the contents of the protocol file.
+    """
+    from aiida.plugins.entry_point import get_entry_point_from_class
+
+    _, entry_point = get_entry_point_from_class(cls.__module__, cls.__name__)
+    entry_point_name = entry_point.name
+    parts = entry_point_name.split('.')
+    parts.pop(0)
+    filename = f'{parts.pop()}.yaml'
+    basepath = functools.reduce(os.path.join, parts)
+
+    with (pathlib.Path(__file__).resolve().parent / basepath / filename).open() as handle:
+        return yaml.safe_load(handle)
+
+
+def get_magnetization_parameters() -> dict:
+    """Return the mapping of suggested initial magnetic moments for each element.
+
+    :returns: the magnetization parameters.
+    """
+    with (pathlib.Path(__file__).resolve().parent / 'magnetization.yaml').open() as handle:
+        return yaml.safe_load(handle)
+
+
+def get_initial_magnetization(structure, pseudo_family):
```

Comment:
Same nitpicky comment as with the `get_builder_from_protocol` methods: perhaps using `starting magnetization` instead of `initial magnetization` is more recognisable for QE users.